### PR TITLE
Text summarization

### LINKED ISSUE #2: Use NLP to Summarize Text
Use NLP to Summarize text from issue #1

### PR DESCRIPTION
**Pull request by KimaruThagna**
# Description
Abstractive text summarization using Torch and Tokenizer
Added text_summary() function

## Related Issues
Closes #2

## Type of Change

- [ ] Bug Fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] The pull request is for branch `develop`
- [ ] `black` gave 0 errors
- [ ] `flake` gave 0 errors
- [ ] Squashed commits into a single one

# Screenshots of working
Screenshot from 2021-02-04 22-23-02

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+transformers==2.8.0
+torch==1.5.0
```
Comment (m0mosenpai) [inline requirements.txt]: I think we can leave the requirements.txt for now. We'll create a separate PR for that + other pre-requisite/dependency/setup.

File: text_summary.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import torch
+from transformers import T5Tokenizer, T5ForConditionalGeneration
+
+model = T5ForConditionalGeneration.from_pretrained('t5-small') # use pretrained model
+tokenizer = T5Tokenizer.from_pretrained('t5-small') # use pretrained tokenizer
+device = torch.device('cpu')
+
+
+def text_summary(text):
```
Comment (m0mosenpai) [inline text_summary.py:9]: We have created a .txt file with the entire captions text inside `videos/<name_of_video>/script.txt`. Look at #11. In the file, I have commented where you need to call your model and where the script file is. So maybe make a function, read the text from that file and then pass it on to the model.

**Review by m0mosenpai — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (KimaruThagna) [reply requirements.txt]: Sure. Will fix this

File: text_summary.py (hunk 2 above)
Comment (KimaruThagna) [reply text_summary.py:9]: Will wait for merges to actualize this. The function will just be a read function. It is independent of my summary function. So I will be implementing it in the clipbit.py file

File: text_summary.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    :return: summarized version of original text
+    '''
+    # preprocessing
+    clean_text = text.strip().replace("\n","")
+    t5_text = "summarize: " + clean_text # define task
+
+    #tokenizer
+    tokenized_text = tokenizer.encode(t5_text, return_tensors="pt").to(device)
+    # summmarize
+    summary_ids = model.generate(tokenized_text,
```
Comment (gmelodie) [inline text_summary.py:22]: Nit: what does `summary_ids` stand for? If you mean "index" I'd suggest `summary_idx` instead

**Review by gmelodie — APPROVED** [review]
LGTM! Just one nitpick but by all means mandatory

**Review by m0mosenpai — CHANGES_REQUESTED** [review]
Squash commits into a single one. Guide:
image

**Pull request merged**